### PR TITLE
Update go.mod in vault/api for internal dependency

### PR DESCRIPTION
**Pull request by michael-bud**

### CODE REVIEW

**Comment by hsimon-hashicorp** [comment]
@michael-bud Is this PR still required? The current version in go.mod is 0.5.3. Is that sufficient for your needs? Please let me know, thanks! :)

**Comment by hsimon-hashicorp** [comment]
Hi, wanted to check back in on this - it looks like currently there's a now a conflict in go.mod, so if you could check on that if you still want this merged in, I'd appreciate it!

File: api/go.mod [hunk 1]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/hashicorp/go-retryablehttp v0.6.6
 	github.com/hashicorp/go-rootcerts v1.0.2
 	github.com/hashicorp/hcl v1.0.0
-	github.com/hashicorp/vault/sdk v0.1.14-0.20200519221838-e0cfd64bc267
+	github.com/hashicorp/vault/sdk v1.5.0
```
Comment (maxb) [inline api/go.mod:15]: This can't be correct ... the current latest version of the sdk is v0.7.0. There is no v1.5.0.

Comment (hsimon-hashicorp) [reply api/go.mod:15]: @maxb that's what I thought too, but also... I'm often wrong. That's what I get for making an assumption the other way! :-D

**Comment by hsimon-hashicorp** [comment]
I'm going to go ahead and close this PR now, after another look at it and because of the time elapsed. Please open a new one if needed. Thanks!

**Pull request closed without merging**